code/modules/vore/eating/bellymodes_vr.dm

```diff
@@ -105,259 +107,195 @@
 						H.bloodstr.add_reagent("numbenzyme",4)
 
 				//Thickbelly flag
-				if(mode_flags & DM_FLAG_THICKBELLY)
-					if(!(H.muffled))
-						H.muffled = 1
+				if((mode_flags & DM_FLAG_THICKBELLY) && !H.muffled)
+					H.muffled = TRUE
 
 				//Stripping flag
 				if(mode_flags & DM_FLAG_STRIPPING)
 					for(var/slot in slots)
 						var/obj/item/I = H.get_equipped_item(slot = slot)
 						if(I)
-							H.unEquip(I,force = FALSE)
-							if(contaminates || istype(I,/obj/item/weapon/card/id))
-								I.gurgle_contaminate(contents, contamination_flavor, contamination_color)
-							if(item_digest_mode == IM_HOLD)
-								items_preserved |= I
-							else if(item_digest_mode == IM_DIGEST_FOOD)
-								if(istype(I,/obj/item/weapon/reagent_containers/food) || istype(I,/obj/item/organ))
-									digest_item(I)
-								else
+							H.unEquip(I, force = FALSE)// We always contaminate IDs.
+							if(contaminates || istype(I, /obj/item/weapon/card/id))
+								I.gurgle_contaminate(H, contamination_flavor, contamination_color)
+
+							switch(item_digest_mode)
+								if(IM_HOLD)
 									items_preserved |= I
-								if(prob(25)) //Less often than with normal digestion
-									if(L && L.client && L.is_preference_enabled(/datum/client_preference/digestion_noises))
-										SEND_SOUND(L,prey_digest)
-									play_sound = pick(pred_digest)
-							else if(item_digest_mode == IM_DIGEST)
-								digest_item(I)
-								if(prob(25)) //Less often than with normal digestion
-									if(L && L.client && L.is_preference_enabled(/datum/client_preference/digestion_noises))
-										SEND_SOUND(L,prey_digest)
-									play_sound = pick(pred_digest)
+								if(IM_DIGEST_FOOD)
+									if(istype(I,/obj/item/weapon/reagent_containers/food) || istype(I,/obj/item/organ))
+										digest_item(I)
+									else
+										items_preserved |= I
+									if(prob(25)) //Less often than with normal digestion
+										if(H && H.client && H.is_preference_enabled(/datum/client_preference/digestion_noises))
+											SEND_SOUND(H, prey_digest)
+										play_sound = pick(pred_digest)
+								if(IM_DIGEST)
+									if(!H && (I.digest_stage && I.digest_stage > 0))
```

Comment:
Oh man I horribly mangled this code because I tried to turn it into a function and didn't actually look back at the main code before trying to revert my change by hand without undoing all of the other changes. Will fix.